### DISCUSSION TITLE
Repeat augmentations with exactly same parameters on two different images

### DISCUSSION THREAD

**Discussion by abhishekaich27** [kornia/kornia#1427] [Questions] [answered]
Hi,

How can I repeat the same transformation (with exactly the same parameters) on two different tensors/images?

For example, if I use `aug = kornia.augmentation.RandomPerspective(0.2, p=1.0)` then, the perspective on two different images `img1` and `img2` should look the same.

**Comment by edgarriba** [comment, marked as answer]
as described in this link https://kornia.readthedocs.io/en/latest/applications/image_augmentations.html#customization , you can do:

```python
img1 = torch.rand(1,3,224,224)
img2 = torch.rand(1,3,224,224)

aug = K.RandomAffine(360, return_transform=True, same_on_batch=True)
img3, trans = aug(img1)
img4 = aug(img2, trans)
```

**Comment by abhishekaich27** [comment]
This doesn't seem to work on `AugmentationSequential`. I did the following:
```
from kornia.augmentation import AugmentationSequential

aug_forms = AugmentationSequential(
    kornia.augmentation.ColorJitter(0.1, 0.1, 0.1, 0.1, p=1.0),
    kornia.augmentation.RandomPerspective(0.2, p=1.0),
    return_transform=True,
    same_on_batch=False)

aug_img, trans = aug_forms(img_1)
aug_img_2 = aug_forms(img_2, trans)
```

This threw the following error:
`AssertionError: The number of inputs must align with the number of data_keys. Got 2 and 1.`

**EDIT**: Corrected a typo.

**Reply by abhishekaich27** [reply]
@edgarriba Any suggestions?

**Reply by edgarriba** [reply]
where is `aug_list` defined ? Notice that in this upcoming release we are deprecating `return_transform` and you will get it with `aug_forms.transform_matrix`. Checkout master for that. /cc @shijianjian

**Reply by abhishekaich27** [reply]
Corrected the typo.  It should be `aug_forms`. 

_Notice that in this upcoming release we are deprecating return_transform and you will get it with aug_forms.transform_matrix_
Thanks. But I have been using the old version with my codebase. Is there a way I can still use the `return_transform` and  address the error I got?

**Reply by shijianjian** [reply]
Reproduce the augmentation is not about transformation matrix. The trans mat is only for geometric transforms. You need the whole set of parameters. Below code shall work.

```python
from kornia.augmentation import AugmentationSequential

aug_forms = AugmentationSequential(
    kornia.augmentation.ColorJitter(0.1, 0.1, 0.1, 0.1, p=1.0),
    kornia.augmentation.RandomPerspective(0.2, p=1.0),
    same_on_batch=False)

aug_img = aug_forms(img_1)
aug_img_2 = aug_forms(img_2, params=aug_forms._params)
```

**Reply by domef** [reply]
Any news about this?

**Reply by edgarriba** [reply]
@abhishekaich27 is the solution from @shijianjian working for you ?
